### ISSUE TITLE
Unexpected PropertyNotSetInConstructor

### ISSUE THREAD

**Issue by mjnbock** [vimeo/psalm#2923]
```php
// PropertyNotSetInConstructor FooBarBaz::$val
class FooBarBaz extends BarBaz{
    public function __construct() { parent::__construct('val'); }
}

class BarBaz extends Baz
{
    /** @var string */
    protected $val;
    //private $val; // no issue found

    public function __construct(string $val='')
    {
        $this->val = $val;
        if (true)
            parent::__construct();

        // no issue found without if:
        //parent::__construct();

        // no issue found without calling parent constructor at all
    }
}

class Baz {
    public function __construct() { }
}
```

This seems to unexpectedly trigger PropertyNotSetInConstructor,
despite the property being set by the parent constructor.

**Comment by muglug** [comment]
Yeah, looks like this was caused by a hacky fix for #1309 – the tests for that still appear to work when I removed the "fix" conditional
